src/vaadin-text-field-mixin.html

```diff
@@ -713,13 +713,17 @@
     }
 
     /**
-     * Returns true if `value` is valid.
-     * `<iron-form>` uses this to check the validity for all its elements.
+     * Validates the field and sets the `invalid` property based on the result.
+     *
+     * The method fires a `validated` event with the result of the validation.
      *
      * @return {boolean} True if the value is valid.
      */
     validate() {
-      return !(this.invalid = !this.checkValidity());
+      const isValid = this.checkValidity();
+      this.invalid = !isValid;
+      this.dispatchEvent(new CustomEvent('validated', {detail: {valid: isValid}}));
+      return isValid;
     }
```

Comment:
It seems we also missed this JSDoc in previous PRs:

https://github.com/vaadin/web-components/pull/4084/files#diff-ce25fa3626d29f442b1b98935ad7c0909865efc4375789d17e1f7b3b18fe392dR34-R40

It has to be added for the event to show up in the API documentation.